### PR TITLE
Create DBContext and Init Controllers and Views

### LINKED ISSUE #27: Create DB context and init Controllers/Views for all models
Generate Controllers/Views for all models
Create DB context
DB init and migration

### PR DESCRIPTION
**Pull request by bsuro10**
#27

### CODE REVIEW

**Review by barBen — DISMISSED** [review]

File: DGN/Models/Article.cs [hunk 1]
```diff
@@ -23,36 +23,31 @@ public class Article
         [DataType(DataType.MultilineText)]
         public string Body { get; set; }
 
-        [DisplayName("The article image location")]
+        [DisplayName("Image name")]
```
Comment (ghost) [inline DGN/Models/Article.cs]: Why?

**Review by ghost — DISMISSED** [review]

File: DGN/Models/Article.cs (hunk 1 above)
Comment (bsuro10) [reply DGN/Models/Article.cs]: Because all the images will share the same path (~/images) 
The only difference is the image name

What do you think?

Comment (ghost) [reply DGN/Models/Article.cs]: Then change the variable name too

Comment (bsuro10) [reply DGN/Models/Article.cs]: I changed it back to location

File: DGN/Models/Article.cs [hunk 2]
```diff
@@ -23,36 +23,31 @@ public class Article
...
         [DisplayName("Last Updated Timestamp")]
         [DataType(DataType.Date)]
         [DisplayFormat(DataFormatString = "{0:dd-MM-yyyy}", ApplyFormatInEditMode = true)]
         public DateTime LastUpdatedTimestamp { get; set; } = DateTime.Now;
 
         public IList<Comment> Comments { get; set; } = new List<Comment>();
 
-        [InverseProperty("ArticleLikes")]
-        public IList<User> Likes { get; set; } = new List<User>();
+        public IList<User> UserLikes { get; set; } = new List<User>();
```
Comment (barBen) [inline DGN/Models/Article.cs:51]: יש לבדוק האם זאת הדרך הנכונה לאתחל את המתשנה הזה.

**Review by ghost — DISMISSED** [review]

File: DGN/Models/Article.cs (hunk 2 above)
Comment (MaayanMordehai) [reply DGN/Models/Article.cs:51]: על איזה משתנה את מדברת, יש פה פשוט כמה?
אם זה הלייקים אז כן רשימה ריקה של יוזרים שעשו לייק למה לא?

**Comment by ghost** [comment]
In the database password table the properties of hash and solt were not saved, so to fix that I added the password constructors according to the following links:
[entity framework with constructors](https://dzone.com/articles/entity-framework-core-supports-constructors-with-a)
[calling constructor from constructor - (look at 5.32)](https://www.informit.com/articles/article.aspx?p=2438407&seqNum=7#:~:text=To%20call%20one%20constructor%20from,constructor%20that%20takes%20two%20parameters.)

**Review by ghost — APPROVED** [review]

File: DGN/Models/Article.cs (hunk 2 above)
Comment (bsuro10) [reply DGN/Models/Article.cs:51]: I checked and It's ok to initialize those fields in that way.
But we need to remember that if we want the controller to create a model **with** the DEFAULT value we need to remove the variable from the form and bind

for example if we want the default value for `Role` when creating a user so we should change:

Create([Bind("Id,Email,Username,Firstname,Lastname,Birthday,**Role**,ImageLocation,About")] User user)

to:
Create([Bind("Id,Email,Username,Firstname,Lastname,Birthday,ImageLocation,About")] User user)

**Review by barBen — APPROVED** [review]

**Pull request merged**